src/server/browserContext.ts

```diff
@@ -283,6 +284,11 @@ export abstract class BrowserContext extends SdkObject {
         // Close all the pages instead of the context,
         // because we cannot close the default context.
         await Promise.all(this.pages().map(page => page.close(metadata)));
+        if (this._browser.options.isChromium) {
```

Comment:
Make ` this._onClosePersistent()` delegate method and implement this in `CRBrowserContext`. Then you don't need a DEPS violation.